addons/barcodes/static/src/barcode_service.js

```diff
@@ -0,0 +1,161 @@
+/** @odoo-module **/
+
+import { isBrowserChrome, isMobileOS } from "@web/core/browser/feature_detection";
+import { registry } from "@web/core/registry";
+import { session } from "@web/session";
+
+const { EventBus, whenReady } = owl;
+
+function isSpecialKey(key) {
+    return (key === "ArrowLeft" || key === "ArrowRight" ||
+        key === "ArrowUp" || key === "ArrowDown" ||
+        key === "Escape" || key === "Tab" ||
+        key === "Backspace" || key === "Delete" ||
+        key === "Home" || key === "End" ||
+        key === "PageUp" || key === "PageDown" ||
+        key === "Shift" || /F\d\d?/.test(key));
+}
+
+function isEditable(element) {
+    return $(element).is('input,textarea,[contenteditable="true"]');
+}
+
+function makeBarcodeInput() {
+    const inputEl = document.createElement('input');
+    inputEl.setAttribute("style", "position:fixed;top:50%;transform:translateY(-50%);z-index:-1;opacity:0");
```

Comment:
Simply a question: why do we put it at 50% from the top and then translate it vertically ?
I don't understand the purpose

Reply:
This doesn't do the same thing: top: 50% is related to the screen height, translateY(-50%) is related to the element's height. By doing this, we put the input with 50% of the screen "as margin-top" but it's too much; we have to reduce it a little bit to be sure to have this input in the middle of the screen. Why we want this for an invisible input?: Because putting focus in field triggers page scrolling on mobile browsers and we wan't to avoid this.

Reply:
Ok I see. I know it's not the same thing but I didn't know for the scrolling on focus, thanks for the explanation :)